python/GafferImageUI/ImageViewUI.py

```diff
@@ -696,35 +747,50 @@ def __updateLabels( self, pixel, colors ) :
 			hovered = Gaffer.Metadata.value( self.getPlug()["area"], "__hovered" )
 		if mode == GafferImageUI.ImageView.ColorInspectorPlug.Mode.Pixel:
 			hovered = Gaffer.Metadata.value( self.getPlug()["pixel"], "__hovered" )
-		prefix = ""
-		postfix = ""
-		if hovered:
-			# Chosen to match brightColor in python/GafferUI/_Stylesheet.py
-			prefix = '<font color="#779cbd">'
-			postfix = '</font>'
-		self.__indexLabel.setText( prefix + ( "" if mode == GafferImageUI.ImageView.ColorInspectorPlug.Mode.Cursor else "<b>" + self.getPlug().getName()[1:] + "</b>" ) + postfix )
-		if mode == GafferImageUI.ImageView.ColorInspectorPlug.Mode.Area:
-			r = self.getPlug()["area"].getValue()
-			self.__positionLabel.setText( prefix + "<b>%i %i -> %i %i</b>" % ( r.min().x, r.min().y, r.max().x, r.max().y ) + postfix )
-		elif mode == GafferImageUI.ImageView.ColorInspectorPlug.Mode.Cursor:
-			self.__positionLabel.setText( prefix + "<b>XY : %i %i</b>" % ( pixel.x, pixel.y ) + postfix )
-		else:
-			p = self.getPlug()["pixel"].getValue()
-			self.__positionLabel.setText( prefix + "<b>XY : %i %i</b>" % ( p.x, p.y ) + postfix )
 
-	def __eventPosition( self, imageGadget, event ):
+		for i in range( 2 ):
+			if i == 1:
+				if sources[0] == sources[1] or self.getPlug().node()["compare"]["mode"].getValue() == "":
+					self.__positionLabels[1].setVisible( False )
+					continue
+				else:
+					self.__positionLabels[1].setVisible( True )
+
+			prefix = ""
+			postfix = ""
+			if i == 0 and hovered:
+				# Chosen to match brightColor in python/GafferUI/_Stylesheet.py
+				prefix = '<font color="#779cbd">'
+				postfix = '</font>'
+
+			self.__indexLabel.setText( prefix + ( "" if mode == GafferImageUI.ImageView.ColorInspectorPlug.Mode.Cursor else "<b>" + self.getPlug().getName()[1:] + "</b>" ) + postfix )
+			if type( sources[i] ) == imath.Box2i:
+				r = sources[i]
+				self.__positionLabels[i].setText( prefix + "<b>%i %i -> %i %i</b>" % ( r.min().x, r.min().y, r.max().x, r.max().y ) + postfix )
+			else:
+				self.__positionLabels[i].setText( prefix + "<b>XY : %i %i</b>" % ( sources[i].x, sources[i].y ) + postfix )
+
+	@staticmethod
+	def __V2iFloor( v ):
+		return imath.V2i( math.floor( v.x ), math.floor( v.y ) )
+
+	def __eventPosition( self, imageGadget, event, floor = True ):
 		try :
 			pixel = imageGadget.pixelAt( event.line )
 		except :
 			# `pixelAt()` can throw if there is an error
 			# computing the image being viewed. We leave
 			# the error reporting to other UI components.
 			return imath.V2i( 0 )
```

Comment:
The intention in this commit seems to be that `__eventPosition` should now return V2f, unless `floor`, in which case V2i is returned. But the error handling here always returns V2i. Is that going to cause problems?

Reply:
I'm not sure it would actually cause user visible problems, but it would definitely be confusing if __pixel was sometimes V2i and sometimes V2f.  Fixed in the squashed commit with all the compare inspector stuff.